File: Loenn/triggers/SubAreaNameTrigger.lua
local SubAreaNameTrigger = {}

SubAreaNameTrigger.name = "XaphanHelper/SubAreaNameTrigger"
SubAreaNameTrigger.fieldOrder = {
    "x", "y", "width", "height", "dialogID", "timer", "textPositionX", "textPositionY"
}
SubAreaNameTrigger.fieldInformation = {
    textPositionX = {
        options = {"Left", "Middle", "Right"},
        editable = false
    },
    textPositionY = {
        fieldType = "integer",
        minimumValue = 80,
        maximumValue = 1080
    }
}
SubAreaNameTrigger.placements = {
    name = "SubAreaNameTrigger",
    data = {
        dialogID = "",
        timer = 3.00,
        textPositionX = "Right",
        textPositionY = 1040
    }
}

return SubAreaNameTrigger

File: Ahorn/triggers/SubAreaNameTrigger.jl
module XaphanHelperSubAreaNameTrigger

using ..Ahorn, Maple

@mapdef Trigger "XaphanHelper/SubAreaNameTrigger" TextTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, dialogID::String="", timer::Number=3.00, textPositionX::String="Right", textPositionY::Integer=1040)

const placements = Ahorn.PlacementDict(
    "Sub-Area Name Trigger (Xaphan Helper)" => Ahorn.EntityPlacement(
        TextTrigger,
        "rectangle"
    )
)

Ahorn.editingOptions(Trigger::TextTrigger) = Dict{String, Any}(
    "textPositionX" => String["Left", "Middle", "Right"]
)

end